treq/client.py

```diff
@@ -86,9 +86,10 @@ def deliverBody(self, protocol):
 
 
 class HTTPClient(object):
-    def __init__(self, agent, cookiejar=None):
+    def __init__(self, agent, cookiejar=None, bodyproducer=IBodyProducer):
         self._agent = agent
         self._cookiejar = cookiejar or cookiejar_from_dict({})
+        self._bodyproducer = bodyproducer
```

Comment:
This attribute should probably be called something more like `dataToBodyProducer` since it is not itself an `IBodyProducer` but rather the way that you get one.